### ISSUE TITLE
Custom module for model training

### ISSUE THREAD

**Issue by LennoxC** [LennoxC/convcnp-assim-nz#16]
More flexibility is required to train on target sets with differing numbers of observations. A custom trainer (outside of what deepsensor provides) is required.

**Comment by LennoxC** [comment]
custom training loop created, using gradient accumulation to handle inputs with different target sizes.
